Top 5 packages based on GitHub Stars (#202)
pharmaverse/blog: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd

### **Features of {admiral}**

Let’s take a look at some features of the package: 

-   [{admiral}](https://github.com/pharmaverse/admiral/) offers **modular functions** for adding variables and records to datasets in a step-by-step manner. This makes it easy to adjust code by adding, removing, or modifying derivations.

-   It provides **template R scripts as a starting point for creating different ADaM datasets.** These templates can be generated using the use_ad_template() function.

-   The package offers a **flexible and efficient way to generate ADaM datasets** from source data, ensuring compliance with CDISC standards and traceability of the data transformation process.

> -   **By providing a consistent set of functions and utilities** for ADaM dataset creation, admiral helps to standardize the reporting process across different clinical trials and pharmaceutical companies.

Reviewer: "admiral" is sometimes within curly brackets and sometimes not, and sometimes it's using a link and sometimes not - i would always use curly brackets and a link.

> -   **Modular design allows for extensions** that focus on specific areas or needs, 3 public packages are currently extending {admiral}, such as [{admiralonco}](https://github.com/pharmaverse/admiralonco), [{admiraloptha}](https://github.com/pharmaverse/admiralophtha) and [{admiralvaccine}](https://github.com/pharmaverse/admiralvaccine). All of them were created from the [admiral template](https://github.com/pharmaverse/admiraltemplate).

Reviewer: ```suggestion
-   **Modular design allows for extensions** that focus on specific areas or needs. Three open-source packages are currently extending {admiral}: [{admiralonco}](https://github.com/pharmaverse/admiralonco), [{admiraloptha}](https://github.com/pharmaverse/admiralophtha) and [{admiralvaccine}](https://github.com/pharmaverse/admiralvaccine). All of them were created from the [admiral template](https://github.com/pharmaverse/admiraltemplate). Two more are in the works: [{admiralpeds}](https://github.com/pharmaverse/admiralpeds) and [{admiralmetabolic}](https://github.com/pharmaverse/admiralmetabolic).
```

  "02", 45, "F", "Treatment", "2023-01-02", "2023-01-17", "Baseline", 1, "2023-01-17", "Pre-dose", "kg", 65, "WEIGHT",
  "03", 54, "F", "", "2022-10-16", "2023-01-09", "Screening", -7, "2023-01-09", "Pre-dose", "kg", 78, "WEIGHT"
)

# Function to aggregate by age group
format_agegr1 <- function(var_input) {
  case_when(
    var_input < 35 ~ "<35",
    between(var_input, 34, 50) ~ "35-49",
    var_input > 50 ~ ">64",
    TRUE ~ "Missing"
  )
}

# Derive ADSL (Subject-Level Analysis Dataset)
adsl <- study_data |>
  select(USUBJID, AGE, SEX, ARM, RANDDTC, RAND2DTC) |>
  distinct() |>
  # Convert blanks strings to NA
  convert_blanks_to_na() |>
  # Admiral does not yet support aggregation function, but dplyr can be used
  mutate(
    AGEGR1 = format_agegr1(AGE)
  ) |>
  # Convert from character to DATE
  derive_vars_dt(
    dtc = RANDDTC,
    new_vars_prefix = "TRTS"
  ) |>
  derive_vars_dt(
    dtc = RAND2DTC,
    new_vars_prefix = "TRTE"
  ) |>
  derive_vars_duration(
    new_var = TRTD,
    start_date = TRTSDT,
    end_date = TRTEDT,
    add_one = FALSE
  )

# Display the ADSL dataset
print(adsl)
```

**Results**

![](images/admiral-example.png)

> **More on Admiral:**

Reviewer: Correct casing on "Admiral"

-   [{admiral} Cheatsheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf)

-   [YouTube playlist of {admiral} videos](https://youtu.be/YO0t8Ul9dEo?si=7hNC8GyuTYh0HWcL)

-   [admiral](https://cran.r-project.org/web/packages/admiral/readme/README.html) 

-   [Get Started • admiral](https://pharmaverse.github.io/admiral/articles/admiral.html#setup) 

-   [Recap: Exploring Clinical Submissions with admiral: An R-Based ADaM Solution with Ben Straub](https://www.appsilon.com/post/exploring-clinical-submissions-with-admiral) \

## **{teal}: Interactive Data Exploration**

![](images/teal.png){width="245"}

> [**{teal}**](https://github.com/insightsengineering/teal)is an open-source R Shiny framework developed by [Roche](https://www.roche.com/) that enables the creation of interactive data exploration applications for the pharmaceutical industry. [{teal}](https://insightsengineering.github.io/teal/latest-tag/) is particularly well-suited for exploring and analyzing data from clinical trials, enabling researchers and clinicians to quickly identify trends, patterns, and insights.\

Reviewer: ```suggestion
[**{teal}**](https://github.com/insightsengineering/teal) is an open-source R Shiny framework developed by [Roche](https://www.roche.com/) that enables the creation of interactive data exploration applications for the pharmaceutical industry. [{teal}](https://insightsengineering.github.io/teal/latest-tag/) is particularly well-suited for exploring and analyzing data from clinical trials, enabling researchers and clinicians to quickly identify trends, patterns, and insights.\
```

-   The package is built on a **modular architecture**, providing a range of pre-built, customizable modules that can be easily integrated into applications to address specific data analysis and visualization needs.

-   It is designed to **seamlessly integrate with other [pharmaverse](https://pharmaverse.org/) packages**, such as [{admiral}](https://github.com/pharmaverse/admiral) and [{rtables}](https://github.com/insightsengineering/rtables).

### **Example**

Here’s the [Patient Profile {teal} application](https://github.com/insightsengineering/teal.gallery/tree/main/patient-profile) for patient-level analysis of clinical trial data from the [teal.gallery](https://insightsengineering.github.io/teal.gallery/).

![](images/teal-example.gif)

**Here’s how you can run the app yourself:**

```
source("https://raw.github.com/insightsengineering/teal.gallery/main/_internal/utils/sourceme.R")

# Run the app
restore_and_run("patient-profile", package_repo = "https://insightsengineering.r-universe.dev")

```

You can also find the [deployed version of the application](https://genentech.shinyapps.io/nest_patient-profile_stable/). 

> **Some Resources on {teal}**

Reviewer: Can you also link the teal blog post you wrote?

```
# Load necessary libraries
library(dplyr)
library(riskmetric)

# Assess and score R packages
pkg_ref(c("riskmetric", "utils", "tools")) %>%
  pkg_assess() %>%
  pkg_score()
```

\
**Results**

![](images/riskassessment-example.png)

It is closely related to [**{riskassessment}**](https://github.com/pharmaR/riskassessment) (the app’s main goal is to help those making “package inclusion” requests for validated GxP environments) and [**{riskscore}**](https://github.com/pharmaR/riskscore) (data package for cataloging riskmetric results across public repositories). \

**More on Riskmetric**

-   [Using {riskassessment} for R Package Validation](https://youtu.be/L0ZZHPh2ec4) 

-   [The {riskmetric} Package](https://www.pharmar.org/riskmetric/) 

-   [GitHub - pharmaR/riskmetric: Metrics to evaluate the risk of R packages](https://github.com/pharmaR/riskmetric) 

## **TidyCDISC: Tidying CDISC Data**

![](images/tidyCDISC.png)

> [**{tidyCDISC}**](https://github.com/Biogen-Inc/tidyCDISC/)is an open-source R package developed by [Biogen](https://www.biogen.com/) that provides a set of functions for tidying and manipulating [CDISC](https://www.cdisc.org/) (Clinical Data Interchange Standards Consortium) datasets. It aims to simplify the process of working with [CDISC](https://www.cdisc.org/) data by providing an intuitive interface for data transformation tasks and ensuring consistency with the principles of tidy data.

Reviewer: ```suggestion
[**{tidyCDISC}**](https://github.com/Biogen-Inc/tidyCDISC/) is an open-source R package developed by [Biogen](https://www.biogen.com/) that provides a set of functions for tidying and manipulating [CDISC](https://www.cdisc.org/) (Clinical Data Interchange Standards Consortium) datasets. It aims to simplify the process of working with [CDISC](https://www.cdisc.org/) data by providing an intuitive interface for data transformation tasks and ensuring consistency with the principles of tidy data.
```

### **Features of TidyCDISC**

1.  [{tidyCDISC}](https://github.com/Biogen-Inc/tidyCDISC/) **adheres to the principles of tidy data**, ensuring that CDISC datasets are structured in a consistent and easy-to-work-with format.

2.  The package **supports various CDISC datasets**, including SDTM (Study Data Tabulation Model), ADaM (Analysis Data Model), and SEND (Standard for Exchange of Nonclinical Data).

> 3.  It provides a set o**f functions for common data transformation tasks,** such as converting between CDISC datasets, handling missing values, and applying CDISC-specific terminology.

Reviewer: ```suggestion
3.  It provides a set of **functions for common data transformation tasks,** such as converting between CDISC datasets, handling missing values, and applying CDISC-specific terminology.
```

-   [{xportr}](https://github.com/atorus-research/xportr) provides **functions to associate metadata information** to an R data frame, apply appropriate types, lengths, labels, formats, and ordering, and then write out a CDISC-compliant XPT file.

-   Performs various **checks on the datasets to ensure CDISC compliance** before exporting to XPT format, including variable name formatting, character length limits, type coercion, and more.

-   Uses a well-defined **specification file** (e.g., ADaM_spec.xlsx) to apply metadata and formatting rules.

-   **Flexible API** allows for the application of transformations individually or chained together in a pipeline.

In summary, [{xportr}](https://github.com/atorus-research/xportr) is a valuable tool for clinical programmers working with CDISC data, as it helps ensure regulatory compliance, data quality, and workflow efficiency when creating XPT files for clinical trials and submissions.

**More on xportr**

-   [xportr 0.4.0](https://pharmaverse.github.io/blog/posts/2024-03-29_xportr_0_4_0/xportr_0_4_0.html)

-   [Utilities to Output CDISC SDTM/ADaM XPT Files • xportr](https://atorus-research.github.io/xportr/) 

## **Conclusion**

> The [pharmaverse](https://pharmaverse.org/) offers a rich ecosystem of tools designed to streamline clinical research workflows, ensuring high-quality data management and reporting. By leveraging packages like [{rtables}](https://github.com/insightsengineering/rtables), [{admiral}](https://github.com/pharmaverse/admiral/tree/main?tab=readme-ov-file), [{teal}](https://github.com/insightsengineering/teal), [{riskmetric}](https://github.com/pharmaR/riskmetric), [{tidyCDISC}](https://github.com/Biogen-Inc/tidyCDISC/), and [{xportr}](https://github.com/atorus-research/xportr/), pharmaceutical companies can enhance their data analysis capabilities, ensure regulatory compliance, and drive innovation in clinical research. **Remember to give the packages a star on GitHub. ⭐**

Reviewer: ```suggestion
The [pharmaverse](https://pharmaverse.org/) offers a rich ecosystem of tools designed to streamline clinical research workflows, ensuring high-quality data management and reporting. By leveraging packages like [{rtables}](https://github.com/insightsengineering/rtables), [{admiral}](https://github.com/pharmaverse/admiral/tree/main?tab=readme-ov-file), [{teal}](https://github.com/insightsengineering/teal), [{riskmetric}](https://github.com/pharmaR/riskmetric), [{tidyCDISC}](https://github.com/Biogen-Inc/tidyCDISC/), and [{xportr}](https://github.com/atorus-research/xportr/), pharmaceutical companies can enhance their data analysis capabilities, ensure regulatory compliance, and drive innovation in clinical research. **Remember to give the packages that you use and value a star on GitHub. ⭐**
```
